packages/openlmis/test/Adaptor.test.js

```diff
@@ -0,0 +1,256 @@
+        method: 'GET',
+        query: {
+          page: 5,
+        },
+      })
+      .reply(200, fixtures.programs);
+
+    const state = {
+      configuration,
+    };
+
+    // prettier-ignore
+    const finalState = await execute(
+      get('programs', { query: { page: 5 }})
+    )(state);
+
+    expect(finalState.data).to.eql(fixtures.programs);
+    expect(finalState.response.statusCode).to.equal(200);
+  });
+
+  it('should get() at /programs with a custom header', async () => {
+    testServer
+      .intercept({
+        path: '/api/programs',
+        method: 'GET',
+        headers: {
+          foo: 'bar',
+        },
+      })
+      .reply(200, fixtures.programs);
+
+    const state = {
+      configuration,
+    };
+
+    // prettier-ignore
+    const finalState = await execute(
+      get('programs', { headers: { 'foo': 'bar' }})
+    )(state);
+
+    expect(finalState.data).to.eql(fixtures.programs);
+    expect(finalState.response.statusCode).to.equal(200);
+  });
+
+  it('should handle a 404', async () => {
+    testServer
+      .intercept({
+        path: '/api/programs/22',
+        method: 'GET',
+      })
+      .reply(
+        404,
+        {
+          message: 'Could not find the resource you were looking for.',
+          code: 404,
+        },
+        {
+          headers: {
+            'content-type': 'application/json',
+          },
+        }
+      );
+
+    const state = {
+      configuration,
+    };
+
+    const error = await get('programs/22')(state).catch(error => {
+      return error;
+    });
+
+    expect(error.statusMessage).to.eql('Not Found');
+    expect(error.statusCode).to.eql(404);
+    expect(error.body.message).to.eql(
+      'Could not find the resource you were looking for.'
+    );
+  });
+
+  it('should post() to /programs', async () => {
+    testServer
+      .intercept({
+        path: '/api/programs',
+        method: 'POST',
+      })
+      .reply(200, {});
+
+    const state = {
+      configuration,
+    };
+
+    const finalState = await execute(
+      post('/programs', {
+        name: 'Program Name',
+        code: 'fine',
+      })
+    )(state);
+
+    expect(finalState.response.statusCode).to.eql(200);
+  });
+
+  it('throws an error when post() returns 400', async () => {
+    testServer
+      .intercept({
+        path: '/api/programs',
+        method: 'POST',
+      })
+      .reply(400, 'Bad Request');
+
+    const state = {
+      configuration,
+    };
+
+    const error = await post('programs', {
+      name: 'Program Name',
+    })(state).catch(error => {
+      return error;
+    });
+
+    expect(error.statusMessage).to.eql('Bad Request');
+    expect(error.message).to.eql(
+      'POST to https://test.openlmis.org/api/programs returned 400: Bad Request'
+    );
+    expect(error.statusCode).to.eql(400);
+  });
+});
+
+describe('Url Matches BaseUrl', () => {
+  it('throw an error when url origin does not match baseUrl', async () => {
```

Comment:
this test should be moved inside the `post` block. It should not be in its own describe